src/Test.UnitTests.Sarif/Core/ResultTests.cs

```diff
@@ -43,6 +48,95 @@ public void Result_Level_ResetsKindValue()
+            Assert.True(WriteSarifThenReadLevelNode(result) == null);
+
+            result = new Result() { Level = FailureLevel.Error };
+            defaultConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Warning };
+            Assert.True(WriteSarifThenReadLevelNode(result, defaultConfiguration) == FailureLevel.Error);
+
+            result = new Result() { Level = FailureLevel.Error };
+            defaultConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Warning };
+            overrideConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Note };
+            Assert.True(WriteSarifThenReadLevelNode(result, defaultConfiguration, overrideConfiguration) == FailureLevel.Error);
+        }
+
+        [Fact]
+        public void Result_Level_WriteCorrectValue_Warning()
+        {
+            Result result;
+            ReportingConfiguration defaultConfiguration;
+            ReportingConfiguration overrideConfiguration;
+
+            result = new Result() { Level = FailureLevel.Warning };
+            Assert.True(WriteSarifThenReadLevelNode(result) == FailureLevel.Warning);
+
+            result = new Result() { Level = FailureLevel.Warning };
+            defaultConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Error };
+            Assert.True(WriteSarifThenReadLevelNode(result, defaultConfiguration) == FailureLevel.Warning);
+
+            result = new Result() { Level = FailureLevel.Warning };
+            defaultConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Error };
+            overrideConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Note };
+            Assert.True(WriteSarifThenReadLevelNode(result, defaultConfiguration, overrideConfiguration) == FailureLevel.Warning);
+        }
+
+        [Fact]
+        public void Result_Level_WriteCorrectValue_NotSet()
+        {
+            Result result;
+            ReportingConfiguration defaultConfiguration;
+            ReportingConfiguration overrideConfiguration;
+
+            result = new Result() { };
+            Assert.True(WriteSarifThenReadLevelNode(result) == null);
+
+            result = new Result() { };
+            defaultConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Error };
+            Assert.True(WriteSarifThenReadLevelNode(result, defaultConfiguration) == null);
+
+            result = new Result() { };
+            defaultConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Error };
+            overrideConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Note };
+            Assert.True(WriteSarifThenReadLevelNode(result, defaultConfiguration, overrideConfiguration) == null);
+        }
+
+        //[Fact]
+        //public void Result_Level_WriteCorrectValue_Null()
```

Comment:
>Null [](http://example.com/codeflow?start=53&length=4)

this test is commented out on purpose, only can be un-commented out with the level change PR, I un-commented out there.